numpy/core/src/multiarray/mapping.c

```diff
@@ -2479,13 +2480,18 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     int i;
     NPY_BEGIN_THREADS_DEF;
 
-    if (mit->size == 0) {
-        /* All indices got broadcast away, do *not* check as it always was */
+    intp_type = PyArray_DescrFromType(NPY_INTP);
+
+    if (NpyIter_GetIterSize(mit->outer) == 0) {
```

Comment:
So this condition means "the size of the broadcast indices"? I'd mistakenly read it as "the size of the non-advanced-index view". Any comment to help clarify that would be great.